zap/tests/selene/checker.rs

```diff
@@ -0,0 +1,52 @@
+use std::{fs, path::PathBuf};
+
+use full_moon::LuaVersion;
+use selene_lib::{standard_library::StandardLibrary, Checker};
+
+pub struct Selene {
+	pub linter: Checker<toml::Value>,
+	pub lua_version: LuaVersion,
+}
+
+fn resolve_standard_library_bases(base_name: &str) -> Vec<StandardLibrary> {
+	let mut resolved_libs: Vec<StandardLibrary> = vec![];
+
+	if let Some(base_library) = StandardLibrary::from_name(base_name) {
+		let base = base_library.base.clone();
+
+		resolved_libs.push(base_library);
+
+		if let Some(next_base) = base {
+			resolved_libs.append(&mut resolve_standard_library_bases(&next_base))
+		}
+	}
+
+	resolved_libs
+}
+
+pub fn initialise_selene() -> Selene {
```

Comment:
This function mimics the way the CLI loads the config/standard libs which is why it's a bit weird